### PR TITLE
feat(common): improve ktlint integration and integrate detekt 

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

File: entity-service/config/detekt/baseline.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" ?>
```
Comment (HoucemKacem) [inline entity-service/config/detekt/baseline.xml:1]: i didn't get it, it's used for what ?

File: entity-service/src/main/kotlin/com/egm/stellio/entity/EntityServiceApplication.kt [hunk 2]
```diff
@@ -8,6 +8,7 @@ import org.springframework.scheduling.annotation.EnableAsync
 @EnableAsync
 class EntityServiceApplication
 
+@Suppress("SpreadOperator")
```
Comment (HoucemKacem) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/EntityServiceApplication.kt:11]: And this tag

File: README.md [hunk 3]
```diff
@@ -110,13 +110,24 @@ If you want to build only one of the services, you can launch:
...
+* You can generate the corresponding ktLint settings with the following command:
+
+```shell script
+./gradlew ktlintApplyToIdea
+```
+
+* You may consider using a plugin like [Save Actions](https://plugins.jetbrains.com/plugin/7642-save-actions) 
+that applies changed code refactoring and optimized imports on a save.
+
+* You can enable Detekt support with the [Detekt plugin](https://github.com/detekt/detekt-intellij-plugin).
```
Comment (HoucemKacem) [inline README.md:130]: Once enabled, how can we perform a check ?

File: entity-service/config/detekt/baseline.xml (hunk 1 above)
Comment (bobeal) [reply entity-service/config/detekt/baseline.xml:1]: it is explained here: https://detekt.github.io/detekt/baseline.html

(we'll fix these existing code smells later, but at least, we won't add new ones)

File: entity-service/src/main/kotlin/com/egm/stellio/entity/EntityServiceApplication.kt (hunk 2 above)
Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/EntityServiceApplication.kt:11]: it disables the spread operator warning for this method (https://detekt.github.io/detekt/performance.html#spreadoperator), which is not really relevant here.

File: README.md (hunk 3 above)
Comment (bobeal) [reply README.md:130]: With the plugin enabled, checks are automatically done and warnings are displayed in the IDE.

File: entity-service/config/detekt/baseline.xml (hunk 1 above)
Comment (HoucemKacem) [reply entity-service/config/detekt/baseline.xml:1]: So having code smells does not break the build

**Review by HoucemKacem — APPROVED** [review]

File: entity-service/config/detekt/baseline.xml (hunk 1 above)
Comment (bobeal) [reply entity-service/config/detekt/baseline.xml:1]: existing code smells won't break it (but we'll have to fix them), new ones will break it.

**Pull request merged**